public/app/plugins/panel/geomap/layers/data/markersLayer.tsx

```diff
@@ -133,31 +134,65 @@ export const markersLayer: MapLayerRegistryItem<MarkersConfig> = {
           ],
         },
       })
-      .addTextInput({
+      .addFieldNamePicker({
         path: 'fieldMapping.latitudeField',
         name: 'Latitude Field',
         defaultValue: defaultOptions.fieldMapping.latitudeField,
+        settings: {
+          filter: (f) => f.type === FieldType.number,
+          noFieldsMessage: 'No numeric fields found',
+        },
         showIf: (config) =>
           config.queryFormat.locationType === 'coordinates',
       })
-      .addTextInput({
+      .addFieldNamePicker({
         path: 'fieldMapping.longitudeField',
         name: 'Longitude Field',
         defaultValue: defaultOptions.fieldMapping.longitudeField,
+        settings: {
+          filter: (f) => f.type === FieldType.number,
+          noFieldsMessage: 'No numeric fields found',
+        },
         showIf: (config) =>
           config.queryFormat.locationType === 'coordinates',
       })
-      .addTextInput({
+      .addFieldNamePicker({
         path: 'fieldMapping.geohashField',
         name: 'Geohash Field',
         defaultValue: defaultOptions.fieldMapping.geohashField,
+        settings: {
+          filter: (f) => f.type === FieldType.string,
+          noFieldsMessage: 'No strings fields found',
+          info: ({
+            name,
+            field,
+          }) => {
+            if(!name || !field) {
+              return <div>Select a field that contains <a href="https://en.wikipedia.org/wiki/Geohash">geohash</a> values in each row.</div>
+            }
+            const first = reduceField({field, reducers:[ReducerID.firstNotNull]})[ReducerID.firstNotNull] as string;
+            if(!first) {
+              return <div>No values found</div>
+            }
+            // const coords = decodeGeohash(first);
+            // if(coords) {
+            //   return <div>First value: {`${coords}`} // {new Date().toISOString()}</div>
+            // }
+            // return <div>Invalid first value: {`${first}`}</div>;
```

Comment:
what were these supposed to be for?

Reply:
I was messing with validating the actual geohash values and showing an error -- I commented it out because it required updates the the geohash function and gt too big :)

The goal is to show clear feedback in the UI when the selected value is OK or not.